### DISCUSSION TITLE
Multiple Devices with same Characteristic Handle

### DISCUSSION THREAD

**Discussion by eric-w-dyn** [hbldh/bleak#724] [Q&A] [answered]
Hello,
I'm working on a project where I need to read information from two separate BLE devices. I followed the example given of setting up notifications from 2 different devices, and am successfully getting data from both. However, I am having a hard time sorting the information between devices as they seem to be given the same characteristic handle. Is there an easy way to sort this? Are the handles assigned by Bleak, or are they determined by the devices themselves?
Thanks for the help!

**Comment by koenvervloesem** [comment, marked as answer]
You can use `functools.partial` to inject the device name into the callback function for the notification (see another discussion about this at https://github.com/hbldh/bleak/discussions/467). Have a look at this example that reads the heart rate from two fitness trackers: https://github.com/hbldh/bleak/issues/630#issuecomment-991877805

**Reply by eric-w-dyn** [reply]
Great, got it working. Cheers
